WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Split Shipments/WooShippingSplitShipmentsViewModel.swift

```
@@ -294,8 +297,16 @@ final class WooShippingSplitShipmentsViewModel: ObservableObject {
     @MainActor
     func saveShipmentInfo() async throws {
         isSavingShipmentInfo = true
-        try await updateShipment()
-        isSavingShipmentInfo = false
+        defer {
+            isSavingShipmentInfo = false
+        }
+
+        do {
+            try await updateShipment()
+        } catch {
+            shouldShowSaveShipmentErrorAlert = true
+            throw error
```

I think we should remove this `throw` if the error is not handled on the view.

---

Keeping the `throw error` allows to call `dismiss` and ` onShipmentUpdate` only in case of success. If error + retry happens, we should keep the screen presented.

```
do {
    try await viewModel.saveShipmentInfo()
    onShipmentUpdate(viewModel.shipments)
    dismiss()
} catch {
    // Error is handled by the ViewModel.
}
```

I'm gonna leave it it as-is.